kairon/actions/definitions/form_validation.py

```diff
@@ -99,3 +103,34 @@ async def execute(self, dispatcher: CollectingDispatcher, tracker: Tracker, doma
             ).save()
 
         return {slot: slot_value}
+
+    async def __execute_http_action(self, action_name: Text, dispatcher: CollectingDispatcher, tracker: Tracker, domain: Dict[Text, Any]):
+        from kairon.actions.definitions.factory import ActionFactory
+
+        action = ActionFactory.get_instance(self.bot, action_name)
+        await action.execute(dispatcher, tracker, domain)
+        if action.is_success:
+            response = action.response
+        return response
+
+    async def __get_slot_value_pre_validation(
+            self, current_value: Any, expected_value: Any, expected_value_type: str,
+            dispatcher: CollectingDispatcher, tracker: Tracker, domain: Dict[Text, Any],
+    ):
+        if expected_value_type == FORM_SLOT_SET_TYPE.custom.value and expected_value is not None:
+            current_value = expected_value
+        elif expected_value_type == FORM_SLOT_SET_TYPE.slot.value:
+            current_value = tracker.get_slot(expected_value)
+        elif expected_value_type == FORM_SLOT_SET_TYPE.action.value:
+            current_value = await self.__execute_http_action(action_name=expected_value, dispatcher=dispatcher,
+                                                             tracker=tracker, domain=domain)
+        return current_value
+
+    def __get_slot_value_post_validation(
+            self, current_value: Any, expected_value: Any, expected_value_type: str, tracker: Tracker,
+    ):
+        if expected_value and expected_value_type == FORM_SLOT_SET_TYPE.custom.value:
+            current_value = expected_value
+        elif expected_value_type == FORM_SLOT_SET_TYPE.slot.value:
+            current_value = tracker.get_slot(expected_value)
+        return current_value
```

Comment:
also add a check: designer should not be able to delete http action linked to form slot set